src/dmd/func.d

```diff
@@ -3088,6 +3088,20 @@ private bool checkEscapingSiblings(FuncDeclaration f, FuncDeclaration outerFunc,
             bAnyClosures = true;
         }
 
+        FuncDeclaration parent = g;
+        while (!bAnyClosures)
+        {
+            parent = parent.parent.isFuncDeclaration;
```

Comment:
This looks a little weird, why not have a for loop?
```
for (auto p = g.parent; p; p = p.parent)
```

Reply:
better?